src/storage/v3/shard_rsm.cpp

```diff
@@ -101,31 +118,239 @@ std::optional<std::map<PropertyId, Value>> CollectAllPropertiesFromAccessor(
+                 [](const auto &prop) { return prop.first; });
+
+  return ret_properties;
+}
+
+std::any ParseExpression(const std::string &expr, memgraph::expr::AstStorage &storage) {
+  memgraph::frontend::opencypher::Parser<memgraph::frontend::opencypher::ParserOpTag::EXPRESSION> parser(expr);
+  ParsingContext pc;
+  CypherMainVisitor visitor(pc, &storage);
+
+  auto *ast = parser.tree();
+  return visitor.visit(ast);
+}
+
+TypedValue ComputeExpression(DbAccessor &dba, const std::optional<memgraph::storage::v3::VertexAccessor> &v_acc,
+                             const std::optional<memgraph::storage::v3::EdgeAccessor> &e_acc,
+                             const std::string &expression, std::string_view node_name, std::string_view edge_name) {
+  AstStorage storage;
+  Frame frame{1 + 1};  // 1 for the node_identifier, 1 for the edge_identifier
+  SymbolTable symbol_table;
+  EvaluationContext ctx;
+
+  ExpressionEvaluator eval{&frame, symbol_table, ctx, &dba, View::OLD};
+  auto expr = ParseExpression(expression, storage);
+
+  auto node_identifier = Identifier(std::string(node_name), false);
+  bool is_node_identifier_present = false;
+  auto edge_identifier = Identifier(std::string(edge_name), false);
+  bool is_edge_identifier_present = false;
+
+  std::vector<Identifier *> identifiers;
+
+  // TODO: use a visitor instead of string search
+  if (v_acc && expression.find(node_name) != std::string::npos) {
+    is_node_identifier_present = true;
+    identifiers.push_back(&node_identifier);
+  }
+  if (e_acc && expression.find(edge_name) != std::string::npos) {
+    is_edge_identifier_present = true;
+    identifiers.push_back(&edge_identifier);
+  }
+
+  expr::SymbolGenerator symbol_generator(&symbol_table, identifiers);
+  (std::any_cast<Expression *>(expr))->Accept(symbol_generator);
+
+  if (is_node_identifier_present) {
+    // We look for the position of node_identifier in the frame.
+    auto position_on_frame = std::find_if(symbol_table.table().begin(), symbol_table.table().end(),
+                                          [&node_identifier](const std::pair<int32_t, Symbol> &position_symbol_pair) {
+                                            return position_symbol_pair.second.name() == node_identifier.name_;
+                                          });
+    MG_ASSERT(position_on_frame != symbol_table.table().end());
+    frame[symbol_table.at(node_identifier)] = *v_acc;
+  }
+  if (is_edge_identifier_present) {
+    // We look for the position of node_identifier in the frame.
+    auto position_on_frame = std::find_if(symbol_table.table().begin(), symbol_table.table().end(),
+                                          [&edge_identifier](const std::pair<int32_t, Symbol> &position_symbol_pair) {
+                                            return position_symbol_pair.second.name() == edge_identifier.name_;
+                                          });
```

Comment:
As you found a proper way to identify whether the edge/node identifier is present, maybe you can eliminate the checks based on strings (line 298 and 302) to make the logic really good. I think you have all the pieces here they just need a bit of restructuring.